Include a `mix ecto` help task similar to hex

joshprice opened the pull request:
Lists the Ecto version, a tagline and the available Ecto mix tasks excluding itself.

```sh
✪ mix ecto
Ecto v2.0.0-rc.5
A database wrapper and language integrated query for Elixir.

Available tasks:

mix ecto.create        # Creates the repository storage
mix ecto.drop          # Drops the repository storage
mix ecto.dump          # Dumps database structure
mix ecto.gen.migration # Generate a new migration for the repo
mix ecto.gen.repo      # Generate a new repository
mix ecto.load          # Loads previously dumped database structure
mix ecto.migrate       # Runs the repository migrations
mix ecto.rollback      # Rolls back migrations
```

Closes #1440

joshprice commented:
I can't figure out why I get this warning on https://github.com/elixir-lang/ecto/pull/1442/files#diff-1d4de435cb6019c16da4599bd29b9c9dR6

```✪ mix test
test/mix/tasks/ecto_test.exs:6: warning: variable version is unused
......
```

whatyouhide commented on test/mix/tasks/ecto_test.exs:
```diff
@@ -0,0 +1,24 @@
+  test "provide help" do
+    Mix.Tasks.Ecto.run []
+    version = "Ecto v#{Ecto.version}"
+    assert_received {:mix_shell, :info, [version]}
```
You're missing a `^` to match on the `version` variable, that's where your warning comes from. :)

josevalim commented on lib/ecto.ex:
```diff
@@ -397,6 +397,8 @@ defmodule Ecto do
 
   """
 
+  def version, do: unquote(Mix.Project.config[:version])
```
There is no need for such, you can run:

```elixir
Application.ensure_all_started(:ecto)
Application.spec(:ecto, :vsn)
```

joshprice replied:
Borrowed that from hex actually https://github.com/hexpm/hex/blob/master/lib/hex.ex#L34

josevalim commented on lib/mix/tasks/ecto.ex:
```diff
@@ -0,0 +1,30 @@
+  @moduledoc """
+  Prints Ecto tasks and their information.
+
+  `mix ecto`
```
Please use four spaces and remove backticks. :)

joshprice replied:
Was wondering whether more docs could go in here to point to more detailed task docs?

josevalim commented:
Thank you, there are some comments and we are good to go! :heart:

joshprice commented:
Rebase or new commit. What do you prefer?

joshprice commented:
Thanks for the feedback everyone. Rebased a new version which:

- remove `Ecto.version` and use `Application.spec` instead
- fixes the variable warning
- improved assertions by simply pattern matching the start of the lines
- fixed formatting in moduledoc

josevalim commented:
:heart: :green_heart: :blue_heart: :yellow_heart: :purple_heart:
